sdks/java/core/src/main/java/org/apache/beam/sdk/coders/GeneratedRowCoder.java

```diff
@@ -0,0 +1,316 @@
+import org.apache.beam.sdk.values.reflect.FieldValueGetter;
+
+/**
+ * A utility for automatically generating a {@link Coder} for {@link Row} objects corresponding to
+ * a specific schema. The resulting coder is loaded into the default ClassLoader and returned.
+ *
+ */
+public abstract class GeneratedRowCoder {
+  private static final ByteBuddy BYTE_BUDDY = new ByteBuddy();
+  private static final ForLoadedType CODER_TYPE = new ForLoadedType(Coder.class);
+  private static final ForLoadedType LIST_CODER_TYPE = new ForLoadedType(ListCoder.class);
+  private static final ForLoadedType MAP_CODER_TYPE = new ForLoadedType(MapCoder.class);
+  private static final BitSetCoder NULL_LIST_CODER = BitSetCoder.of();
+
+  public static final String CODERS_FIELD_NAME = "FIELD_CODERS";
+
+  // A map of primitive types -> StackManipulations to create their coders.
+  private static final Map<TypeName, StackManipulation> CODER_MAP;
+
+  // Cache for Coder class that are already generated.
+  private static Map<UUID, Coder<Row>> generatedCoders = Maps.newHashMap();
+
+  static {
+    // Initialize the CODER_MAP with the StackManipulations to create the primitive coders.
+    // Assumes that each class contains a static of() constructor method.
+    CODER_MAP = Maps.newHashMap();
+    for (Map.Entry<TypeName, Coder> entry : RowCoder.CODER_MAP.entrySet()) {
+      StackManipulation stackManipulation = MethodInvocation.invoke(
+          new ForLoadedType(entry.getValue().getClass()).getDeclaredMethods()
+              .filter(ElementMatchers.named(("of")))
+              .getOnly());
+      CODER_MAP.putIfAbsent(entry.getKey(), stackManipulation);
+    }
+  }
+
+  public static Coder<Row> of(Schema schema, UUID coderId) {
+    return generatedCoders.computeIfAbsent(coderId,
+        h -> {
+          TypeDescription.Generic coderType =
+              TypeDescription.Generic.Builder.parameterizedType(Coder.class, Row.class).build();
+          DynamicType.Builder<Coder> builder =
+              (DynamicType.Builder<Coder>) BYTE_BUDDY.subclass(coderType);
+          builder = createComponentCoders(schema, builder);
+          builder = implementMethods(builder, schema);
+          try {
+            return builder
+                .make()
+                .load(
+                    Coder.class.getClassLoader(),
+                    ClassLoadingStrategy.Default.INJECTION)
+                .getLoaded()
+                .getDeclaredConstructor().newInstance();
+          } catch (InstantiationException | IllegalAccessException
+              | NoSuchMethodException | InvocationTargetException e) {
+            throw new RuntimeException("Unable to generate coder for schema " + schema);
+          }
+        });
+  }
+
+  static DynamicType.Builder<Coder> implementMethods(
+      DynamicType.Builder<Coder> builder, Schema schema) {
+    return builder
+        .defineMethod("getSchema", Schema.class, Visibility.PRIVATE, Ownership.STATIC)
```

Comment:
Does the ownership have to be Static? IMO, this method is only called during decoding, which is called inside from the instance.